### PR TITLE
Access request compatibility for servers without v2 api

### PR DESCRIPTION
**Pull request by xacrimon**
Fixes a compatability issue introduced by security fixes where the client would call a grpc method that doesn't exist.

### CODE REVIEW

File: api/client/client.go [hunk 1]
```diff
@@ -783,6 +783,11 @@ func (c *Client) GetBotUsers(ctx context.Context) ([]types.User, error) {
 func (c *Client) GetAccessRequests(ctx context.Context, filter types.AccessRequestFilter) ([]types.AccessRequest, error) {
 	stream, err := c.grpc.GetAccessRequestsV2(ctx, &filter, c.callOpts...)
 	if err != nil {
+		err := trail.FromGRPC(err)
+		if trace.IsNotImplemented(err) {
```
Comment (codingllama) [inline api/client/client.go:787]: nit: overshadow in a smaller scope?

```suggestion
		if err := trail.FromGRPC(err); trace.IsNotImplemented(err) {
```

File: api/client/client.go [hunk 2]
```diff
@@ -800,6 +805,23 @@ func (c *Client) GetAccessRequests(ctx context.Context, filter types.AccessReque
 	return reqs, nil
 }
 
+// getAccessRequestsLegacy retrieves a list of all access requests matching the provided filter using the old access request API.
+//
+// DELETE IN: 11.0.0
```
Comment (codingllama) [inline api/client/client.go:810]: nit:

```suggestion
// DELETE IN 11.0.0. <Insert brief explanation> (xacrimon).
```

**Review by codingllama — COMMENTED** [review]
Could we add a test for this?

**Review by r0mant — APPROVED** [review]
lgtm once @codingllama's comments are addressed

File: api/client/client.go (hunk 1 above)
Comment (xacrimon) [reply api/client/client.go:787]: Removed the last conversion instead to avoid doing it twice.

**Comment by xacrimon** [comment]
@codingllama Please take another look now

**Comment by codingllama** [comment]
Hey @xacrimon, the main ask here was if we could add a test for this. WDYT?

**Comment by xacrimon** [comment]
@codingllama Ah, the emails I got from GitHub had that first comment collapsed. If what you're asking is if we can precisely test the compat to make sure it hits the fallback. I'm not sure on how to do that to be honest, you'd have to somehow mock gRPC into thinking a method is unimplemented (otherwise we'd end up with a lot of wacky interfaces and indirections everywhere), I'm not aware of any other similar tests in Teleport but if you have any ideas I'm all ears.

**Review by codingllama — APPROVED** [review]

**Comment by codingllama** [comment]
Approving, as it seems unfair to hold you on something we don't seem to do. We can do a follow up if you like.

> if you have any ideas I'm all ears.

I think creating a new gprc.Server and binding proto.UnimplementedAuthServiceServer to it would do it.

**Review by marcoandredinis — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**